src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.cs

```diff
@@ -2,314 +2,61 @@
-            bool looksGeneric, INamedTypeSymbol searchResult)
-        {
-            if (arity != 0 && searchResult.GetArity() != arity)
-            {
-                // If the user supplied type arguments, then the search result has to match the 
-                // number provided.
-                return false;
-            }
-
-            if (looksGeneric && searchResult.TypeArguments.Length == 0)
-            {
-                return false;
-            }
-
-            if (!searchResult.IsAccessibleWithin(semanticModel.Compilation.Assembly))
-            {
-                // Search result has to be accessible from our current location.
-                return false;
-            }
-
-            if (inAttributeContext && !searchResult.IsAttribute())
-            {
-                // If we need an attribute, we have to have found an attribute.
-                return false;
-            }
+            var service = document.GetRequiredLanguageService<IFullyQualifyService>();
+            var optFixData = await service.GetFixDataAsync(document, context.Span, hideAdvancedMembers, cancellationToken).ConfigureAwait(false);
+            if (optFixData is null)
+                return;
 
-            if (!HasValidContainer(searchResult))
-            {
-                // Named type we find must be in a namespace, or a non-generic type.
-                return false;
-            }
+            var fixData = optFixData.Value;
+            if (fixData.IndividualFixData.Length == 0)
+                return;
 
-            return true;
-        }
+            var codeActions = fixData.IndividualFixData.SelectAsArray(
+                d => CodeAction.Create(
+                    d.Title,
+                    async cancellationToken =>
+                    {
+                        var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var newText = sourceText.WithChanges(d.TextChanges);
+                        return document.WithText(newText);
+                    },
+                    d.Title));
 
-        private static bool HasValidContainer(ISymbol symbol)
-            => symbol.ContainingSymbol is INamespaceSymbol or INamedTypeSymbol { IsGenericType: false };
-
-        private bool BindsWithoutErrors(INamespaceSymbol ns, string? rightName, bool isAttributeName)
-        {
-            // If there was no name on the right, then this binds without any problems.
-            if (rightName == null)
-                return true;
-
-            // Otherwise, see if the namespace we will bind this contains a member with the same
-            // name as the name on the right.
-            var types = ns.GetMembers(rightName);
-            if (types.Any())
-                return true;
-
-            if (!isAttributeName)
+            if (codeActions.Length >= 2)
             {
-                return false;
+                // Wrap the spell actions into a single top level suggestion
```

Comment:
spell?

Reply:
Looks like a bogus comment in the original as well :+). Will fix